apps/website/src/pages/giveaways/[giveawayId]/rules.tsx

```diff
@@ -6,33 +6,21 @@ import type {
 } from "next";
 
 import type { Giveaway } from "@prisma/client";
-import { prisma } from "@/server/db/client";
 
 import Heading from "@/components/content/Heading";
 import Section from "@/components/content/Section";
 import Meta from "@/components/content/Meta";
 import { calcGiveawayConfig } from "@/utils/giveaways";
-
-async function findActiveGiveaway(giveawaySlugOrId: string) {
-  const now = new Date();
-  return await prisma.giveaway.findFirst({
-    where: {
-      active: true,
-      startAt: { lt: now },
-      AND: [
-        { OR: [{ endAt: null }, { endAt: { gt: now } }] },
-        { OR: [{ id: giveawaySlugOrId }, { slug: giveawaySlugOrId }] },
-      ],
-    },
-  });
-}
+import { findActiveGiveaway } from "@/server/db/giveaways";
+import Markdown from "@/components/content/Markdown";
 
 export type GiveawayPageProps = InferGetServerSidePropsType<
   typeof getServerSideProps
 >;
 
 export const getServerSideProps: GetServerSideProps<{
```

Comment:
Maybe not for this PR, but are we still thinking about caching for this?

Reply:
Yeah I have a few things I want to add to giveaways like a gdpr/privacy policy checkbox, but wanted to get this PR done first.